DataDog/datadog-lambda-js
# Add DynamoDB Downstream Span Pointers

### What does this PR do?

Adds span pointers to spans for Lambdas triggered by DynamoDB INSERT, MODIFY, and REMOVE event streams. In practice, this covers DynamoDB updates caused by `putItem`, `updateItem`, `deleteItem`, `transactWriteItems`, and `batchWriteItems` requests.

This also updates the span pointer util imports that were modified in https://github.com/DataDog/dd-trace-js/pull/4912.

<img width="1334" alt="Screenshot 2024-12-18 at 2 36 57 PM" src="https://github.com/user-attachments/assets/52985cb5-409a-45a5-9dc2-828da447eb01" />
<img width="1345" alt="Screenshot 2024-12-18 at 2 38 24 PM" src="https://github.com/user-attachments/assets/4b3c97ad-e07b-4f05-a3eb-803ba4f97736" />
<img width="1342" alt="Screenshot 2024-12-18 at 2 41 02 PM" src="https://github.com/user-attachments/assets/93f11b95-38a9-464d-a56e-107c11801235" />

### Motivation

This feature already exists in Python, and I'm working on adding it to all other runtimes (Node, .NET, Java, Golang).

### Testing Guidelines

Easy: Checkout [this span](https://ddserverless.datadoghq.com/apm/trace/1500157059434337211?config_trace_show_hidden_metadata=true&graphType=flamegraph&shouldShowLegend=true&sort=time&spanID=5241261389336113523&spanViewType=span-links&timeHint=1734556832596.0056) enable the feature flag, and you will see that it's pointing to the downstream Lambda.

More thorough: Run [this Lambda function](https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/nhulston-javascript-upstream-dynamo-dev-hello?subtab=envVars&tab=testing) with the event payload

```json
{
  "shouldUseV2": false,
  "twoKeyTable": false,
  "shouldPutItem": false,
  "shouldUpdateItem": false,
  "shouldDeleteItem": false,
  "shouldBatchWriteItems": false,
  "shouldTransactWriteItems": false
}
```
As you can see, there are a lot of different cases to test -- v2 vs v3 AWS SDK, one primary key vs two primary key tables, and the 5 different types of requests. You can test by enabling one of the requests, running the Lambda, and then check Datadog to ensure that the spans are properly linked.

### Additional Notes

- This handles the 'downstream' case, for Lambdas that are triggered by DynamoDB updates. https://github.com/DataDog/dd-trace-js/pull/4912 adds span pointers on the upstream case (for Lambdas that make the DynamoDB requests)
- Tests will fail until dd-trace-js is released and we bump the version in datadog-lambda-js first.

### Types of Changes


### Check all that apply

- [ ] This PR's changes are covered by the automated tests -- todo, will add to self-monitoring
